sound/soc/codecs/hdac_hda.c

```diff
@@ -621,9 +631,15 @@ static int hdac_hda_dev_probe(struct hdac_device *hdev)
 	snd_hdac_ext_bus_link_get(hdev->bus, hlink);
 
 	/* ASoC specific initialization */
-	ret = devm_snd_soc_register_component(&hdev->dev,
-					 &hdac_hda_codec, hdac_hda_dais,
-					 ARRAY_SIZE(hdac_hda_dais));
+	if (snd_hda_device_is_hdmi(hdev))
```

Comment:
it is Intel only, (AMD's HDMI is addr=0 in some cases for example).

I'm looking at a different method, something like this:
hcodec = hdac_to_hda_codec(hdac);
and from there try to find some way. So far no leads in there.
There is a guessing local function in hda_bind.c: `is_likely_hdmi_codec()` but that might be even more questionable.

I'll try to find a better way or make this a bit nicer.

Reply:
To clarify: I'm not sure about addr == 2, I just only saw it in code for Intel platforms, it might be universal HDA convention, but I'm not certain if it is correct or not